api/src/main/java/org/apache/iceberg/ScanTaskGroup.java

```diff
@@ -26,6 +26,24 @@
  * @param <T> the type of scan tasks
  */
 public interface ScanTaskGroup<T extends ScanTask> extends ScanTask {
+  /**
+   * Returns a grouping key for this task group.
+   *
+   * <p>Tasks may produce records that belong to a specific group of data (e.g. partition). It may
```

Comment:
This description is a little too abstract for me to understand what the grouping key is from this doc. I think grouping key needs to be defined first here. Maybe 

"A grouping key is a set of values that are common amongst all rows within all tasks within the group. The values may be the result of transforming the underlying data. For example, a grouping key may have the bucket ordinal computed by applying a bucketing transformation to a column of the underlying rows."

Wdyt?

Reply:
I agree. Changed.